<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MuSiC: Multi-subject Single-cell Deconvolution • MuSiC</title>
<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MuSiC: Multi-subject Single-cell Deconvolution">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MuSiC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/MuSiC.html">Tutorial</a>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li>
  <a href="../articles/pages/data.html">Data</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/xuranw/MuSiC">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>MuSiC: Multi-subject Single-cell Deconvolution</h1>
                        <h4 class="author">Xuran Wang</h4>
            <address class="author_afil">
      Graduate Group of AMCS, University of Pennsylvania, Philadelphia, USA<br><small class="dont-index">Source: <a href="https://github.com/xuranw/MuSiC/blob/master/vignettes/MuSiC.Rmd"><code>vignettes/MuSiC.Rmd</code></a></small>
      <div class="hidden name"><code>MuSiC.Rmd</code></div>

    </address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette provides a walk through tutorial on how to use <code>MuSiC</code> to estimate cell type proportions from bulk sequencing data based on multi-subject single cell data by reproducing the analysis in MuSiC <a href="https://www.biorxiv.org/content/early/2018/06/26/354944">manuscript</a>.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To run the entire deconvolution tutorial, users need to install the <a href="https://www.github.com/xuranw/MuSiC"><code>MuSiC</code></a> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install devtools if necessary</span>
<span class="kw">install.packages</span>(<span class="st">'devtools'</span>)

<span class="co"># install the MuSiC package</span>
devtools::<span class="kw">install_github</span>(<span class="st">'xuranw/MuSiC'</span>)

<span class="co"># load</span>
<span class="kw">library</span>(MuSiC)</code></pre></div>
</div>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h2>
<p><code>MuSiC</code> uses two types of input data:</p>
<ul>
<li><p>Bulk expression obtained from RNA sequencing, which is a mixture expression of various cell types. These are the data we want to deconvolve.</p></li>
<li><p>Multi-subject single cell expression obtained from single-cell RNA sequencing (scRNA-seq). The cell types of scRNA-seq are pre-determined. These serve as reference for estimating cell type proportions of bulk data.</p></li>
</ul>
<p><code>MuSiC</code> requires raw read counts for both bulk and single-cell expression. The discussion of the usage of RPKM and TPM can be found in the <strong>Discussion</strong> section of our <a href="https://www.biorxiv.org/content/early/2018/06/26/354944">manuscript</a>.</p>
</div>
</div>
<div id="music-deconvolution" class="section level1">
<h1 class="hasAnchor">
<a href="#music-deconvolution" class="anchor"></a>MuSiC Deconvolution</h1>
<p>MuSiC utilizes cell-type specific gene expression from single-cell RNA sequencing (RNA-seq) data to characterize cell type compositions from bulk RNA-seq data in complex tissues. By appropriate weighting of genes showing cross-subject and cross-cell consistency, MuSiC enables the transfer of cell type-specific gene expression information from one dataset to another.</p>
<p>Solid tissues often contain closely related cell types which leads to collinearity. To deal with collinearity, MuSiC employs a tree-guided procedure that recursively zooms in on closely related cell types. Briefly, we first group similar cell types into the same cluster and estimate cluster proportions, then recursively repeat this procedure within each cluster.</p>
<div class="figure">
<img src="images/FigureMethod.jpg">
</div>
</div>
<div id="sample-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#sample-analysis" class="anchor"></a>Sample Analysis</h1>
<div id="data-summary" class="section level2">
<h2 class="hasAnchor">
<a href="#data-summary" class="anchor"></a>Data Summary</h2>
<p>This vignette reproduces the <strong>human pancreatic islet</strong> and the <strong>mouse kidney</strong> analysis, which require single cell and bulk RNA-seq datasets from following sources:</p>
<table class="table">
<colgroup>
<col width="13%">
<col width="23%">
<col width="13%">
<col width="20%">
<col width="17%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th>Datasets</th>
<th><span class="citation">Segerstolpe et al. (2016)</span></th>
<th><span class="citation">Xin et al. (2016)</span></th>
<th><span class="citation">Fadista et al. (2014)</span></th>
<th><span class="citation">J. Park et al. (2018)</span></th>
<th><span class="citation">Beckerman et al. (2017)</span></th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Session Num.</td>
<td><a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-5061/">E-MTAB-5061</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81608">GSE81608</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50244">GSE50244</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107585">GSE107585</a></td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81492">GSE81492</a></td>
</tr>
<tr class="even">
<td>Data type</td>
<td>scRNA-seq</td>
<td>scRNA-seq</td>
<td>bulk RNA-seq</td>
<td>scRNA-seq</td>
<td>bulk RNA-seq</td>
</tr>
<tr class="odd">
<td>Protocol</td>
<td>Smart-seq2</td>
<td>Illumina HiSeq 2500</td>
<td>Illumina HiSeq 2000</td>
<td>10x</td>
<td>Illumina HiSeq 2500</td>
</tr>
<tr class="even">
<td>Tissue</td>
<td><strong>human pancreas</strong></td>
<td><strong>human pancreas</strong></td>
<td><strong>human pancreas</strong></td>
<td><strong>mouse kidney</strong></td>
<td><strong>mouse kidney</strong></td>
</tr>
<tr class="odd">
<td># of genes</td>
<td>25,453</td>
<td>39,849</td>
<td>56,638</td>
<td>16,273</td>
<td>19,033</td>
</tr>
<tr class="even">
<td># of cells</td>
<td>2,209</td>
<td>1,492</td>
<td>NA</td>
<td>43,745</td>
<td>NA</td>
</tr>
<tr class="odd">
<td># of cell types</td>
<td>14 + 1 NA</td>
<td>4</td>
<td>NA</td>
<td>14 + 2 novel</td>
<td>NA</td>
</tr>
<tr class="even">
<td># of subjects</td>
<td>10 (6 healthy + 4 T2D)</td>
<td>18 (12 healthy + 6 T2D)</td>
<td>89</td>
<td>7</td>
<td>10 (6 control + 4 APOL1)</td>
</tr>
</tbody>
</table>
<p>Bioconduct base package provides <code>ExpressionSet</code> class, which is a convenient data structure to hold expression data along with sample/feature annotation. Here we use two <code>ExpressionSet</code> objects to handle the bulk and single cell data respectively. The details of constructing <code>ExpressionSet</code> can be found on <a href="https://www.rdocumentation.org/packages/Biobase/versions/2.32.0/topics/ExpressionSet">this page</a>.</p>
<p>Datasets described in the table above are all in the form of <code>ExpressionSet</code> and available at the <a href="pages/data.html">data download page</a>.</p>
</div>
<div id="estimation-of-cell-type-proportions" class="section level2">
<h2 class="hasAnchor">
<a href="#estimation-of-cell-type-proportions" class="anchor"></a>Estimation of cell type proportions</h2>
<p>In stead of selecting marker genes, MuSiC gives weights to each gene. The weight scheme is based on cross-subject variation: up-weight gene with low variation and down-weight gene with high variation. The composition of bulk tisue samples from there total gene expression is inferred by both cross-subject mean and variance by all genes. Here we domenstrate step by step with human pancreas datasets.</p>
<div id="data-preparations" class="section level3">
<h3 class="hasAnchor">
<a href="#data-preparations" class="anchor"></a>Data Preparations</h3>
<div id="bulk-data-of-human-pancreas" class="section level4">
<h4 class="hasAnchor">
<a href="#bulk-data-of-human-pancreas" class="anchor"></a>Bulk data of human pancreas</h4>
<p>The dataset from <span class="citation">Fadista et al. (2014)</span> contains raw read counts data on global transcriptomic analysis of bulk human pancreatic islets to study glucose metabolism in healthy and hyper-glycemic conditions. For the purpose of this vignette, the dataset is pre-propocessed and made available on the <a href="pages/data.html">data download page</a>. In addition to read counts, this dataset also contains HbA1c levels, BMI, gender and age information for subjects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">GSE50244.bulk.eset =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">'https://xuranw.github.io/MuSiC/data/GSE50244bulkeset.rds'</span>)
GSE50244.bulk.eset
<span class="co">#ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">#assayData: 32581 features, 89 samples </span>
<span class="co">#  element names: exprs </span>
<span class="co">#protocolData: none</span>
<span class="co">#phenoData</span>
<span class="co">#  sampleNames: Sub1 Sub2 ... Sub89 (89 total)</span>
<span class="co">#  varLabels: sampleID SubjectName ... tissue (7 total)</span>
<span class="co">#  varMetadata: labelDescription</span>
<span class="co">#featureData: none</span>
<span class="co">#experimentData: use 'experimentData(object)'</span>
<span class="co">#Annotation:  </span></code></pre></div>
</div>
<div id="single-cell-data-of-human-pancreas" class="section level4">
<h4 class="hasAnchor">
<a href="#single-cell-data-of-human-pancreas" class="anchor"></a>Single cell data of human pancreas</h4>
<p>The single cell data are from <span class="citation">Segerstolpe et al. (2016)</span>, which contrains read counts for 25453 genes of 2209 cells. Here we only include the 1097 cell from 6 healthy subjects. The read counts are available on the <a href="pages/data.html">data download page</a>, in the form of an <code>ExpressionSet</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Download EMTAB single cell dataset from Github</span>
EMTAB.eset =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">'https://xuranw.github.io/MuSiC/data/EMTABesethealthy.rds'</span>)
EMTAB.eset
<span class="co">#ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">#assayData: 25453 features, 1097 samples </span>
<span class="co">#  element names: exprs </span>
<span class="co">#protocolData: none</span>
<span class="co">#phenoData</span>
<span class="co">#  sampleNames: AZ_A10 AZ_A11 ... HP1509101_P9 (1097 total)</span>
<span class="co">#  varLabels: sampleID SubjectName cellTypeID cellType</span>
<span class="co">#  varMetadata: labelDescription</span>
<span class="co">#featureData: none</span>
<span class="co">#experimentData: use 'experimentData(object)'</span>
<span class="co">#Annotation:  </span></code></pre></div>
<p>Another single cell data are from <span class="citation">Xin et al. (2016)</span>, which have 39849 genes and 1492 cells. The read counts are available on the <a href="pages/data.html">data download page</a>, in the form of an <code>ExpressionSet</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Download Xin et al. single cell dataset from Github</span>
XinT2D.eset =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">'https://xuranw.github.io/MuSiC/data/XinT2Deset.rds'</span>)
XinT2D.eset
<span class="co">#ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">#assayData: 39849 features, 1492 samples </span>
<span class="co">#  element names: exprs </span>
<span class="co">#protocolData: none</span>
<span class="co">#phenoData</span>
<span class="co">#  sampleNames: Sample_1 Sample_2 ... Sample_1492 (1492 total)</span>
<span class="co">#  varLabels: sampleID SubjectName ... Disease (5 total)</span>
<span class="co">#  varMetadata: labelDescription</span>
<span class="co">#featureData: none</span>
<span class="co">#experimentData: use 'experimentData(object)'</span>
<span class="co">#Annotation:  </span></code></pre></div>
</div>
</div>
<div id="benchmark-evaluation" class="section level3">
<h3 class="hasAnchor">
<a href="#benchmark-evaluation" class="anchor"></a>Benchmark Evaluation</h3>
<p>Benchmark dataset is constructed by summing up single cell data from <code>XinT2D.eset</code>. The artificial bulk data is constructed through function <a href="../reference/bulk_construct.html"><code>bulk_construct</code></a>. The inputs are single cell dataset, cluster name (<code>clusters</code>), sample name (<code>samples</code>) and selected cell type (<code>select.ct</code>). <a href="../reference/bulk_construct.html"><code>bulk_construct</code></a> returns a ExpressionSet of artificial bulk dataset <code>Bulk.counts</code> and a matrix of real cell type counts <code>num.real</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Construct artificial bulk dataset. Use all 4 cell types: alpha, beta, gamma, delta</span>
XinT2D.construct.full =<span class="st"> </span><span class="kw"><a href="../reference/bulk_construct.html">bulk_construct</a></span>(XinT2D.eset, <span class="dt">clusters =</span> <span class="st">'cellType'</span>, <span class="dt">samples =</span> <span class="st">'SubjectName'</span>)

<span class="kw">names</span>(XinT2D.construct.full)
<span class="co"># [1] "Bulk.counts" "num.real" </span>

XinT2D.construct.full$Bulk.counts
<span class="co">#ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">#assayData: 39849 features, 18 samples </span>
<span class="co">#element names: exprs </span>
<span class="co">#protocolData: none</span>
<span class="co">#phenoData</span>
<span class="co">#sampleNames: Non T2D 1 Non T2D 2 ... T2D 6 (18 total)</span>
<span class="co">#varLabels: SubjectName sampleID Disease</span>
<span class="co">#varMetadata: labelDescription</span>
<span class="co">#featureData: none</span>
<span class="co">#experimentData: use 'experimentData(object)'</span>
<span class="co">#Annotation:  </span>

<span class="kw">head</span>(XinT2D.construct.full$num.real)
<span class="co">#          alpha beta delta gamma</span>
<span class="co">#Non T2D 1    53   13     5     3</span>
<span class="co">#Non T2D 2     4   13     2     5</span>
<span class="co">#Non T2D 3    27   10     3     2</span>
<span class="co">#Non T2D 4    14   10     0     3</span>
<span class="co">#Non T2D 5    23   22     5     2</span>
<span class="co">#Non T2D 6    36    7     4     1</span>

<span class="co"># calculate cell type proportions</span>
XinT2D.construct.full$prop.real =<span class="st"> </span><span class="kw"><a href="../reference/relative.ab.html">relative.ab</a></span>(XinT2D.construct.full$num.real, <span class="dt">by.col =</span> <span class="ot">FALSE</span>)
<span class="kw">head</span>(XinT2D.construct.full$prop.real)
<span class="co">#              alpha      beta      delta      gamma</span>
<span class="co">#Non T2D 1 0.7162162 0.1756757 0.06756757 0.04054054</span>
<span class="co">#Non T2D 2 0.1666667 0.5416667 0.08333333 0.20833333</span>
<span class="co">#Non T2D 3 0.6428571 0.2380952 0.07142857 0.04761905</span>
<span class="co">#Non T2D 4 0.5185185 0.3703704 0.00000000 0.11111111</span>
<span class="co">#Non T2D 5 0.4423077 0.4230769 0.09615385 0.03846154</span>
<span class="co">#Non T2D 6 0.7500000 0.1458333 0.08333333 0.02083333</span></code></pre></div>
<p>The cell type proportions is estimated by function <a href="../reference/music_prop.html"><code>music_prop</code></a>. We can specify the way of grouping scRNA-seq data by <code>cluster</code>. <code>samples</code> refers to different subjects and <code>select.ct</code> refers to cell type included (use all cell types in the single cell dataset if not specified). The estimated proportions are normalized based on included cell type.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(xbioc)

<span class="co"># Estimate cell type proportions of artificial bulk data</span>
Est.prop.Xin =<span class="st"> </span><span class="kw"><a href="../reference/music_prop.html">music_prop</a></span>(<span class="dt">bulk.eset =</span> XinT2D.construct.full$Bulk.counts, <span class="dt">sc.eset =</span> EMTAB.eset,
                          <span class="dt">clusters =</span> <span class="st">'cellType'</span>, <span class="dt">samples =</span> <span class="st">'sampleID'</span>, 
                          <span class="dt">select.ct =</span> <span class="kw">c</span>(<span class="st">'alpha'</span>, <span class="st">'beta'</span>, <span class="st">'delta'</span>, <span class="st">'gamma'</span>))
<span class="co">#Creating Relative Abudance Matrix...</span>
<span class="co">#Creating Variance Matrix...</span>
<span class="co">#Creating Library Size Matrix...</span>
<span class="co">#Used 17884 common genes...</span>
<span class="co">#Used 4 cell types in deconvolution...</span>
<span class="co">#Non T2D 1 has common genes 15115 ...</span>
<span class="co">#Non T2D 2 has common genes 12826 ...</span>
<span class="co">#Non T2D 3 has common genes 13553 ...</span>
<span class="co">#Non T2D 4 has common genes 13491 ...</span>
<span class="co">#Non T2D 5 has common genes 14585 ...</span>
<span class="co">#Non T2D 6 has common genes 13984 ...</span>
<span class="co">#Non T2D 7 has common genes 13065 ...</span>
<span class="co">#Non T2D 8 has common genes 13756 ...</span>
<span class="co">#Non T2D 9 has common genes 13813 ...</span>
<span class="co">#Non T2D 10 has common genes 14566 ...</span>
<span class="co">#Non T2D 11 has common genes 14197 ...</span>
<span class="co">#Non T2D 12 has common genes 15507 ...</span>
<span class="co">#T2D 1 has common genes 14936 ...</span>
<span class="co">#T2D 2 has common genes 15084 ...</span>
<span class="co">#T2D 3 has common genes 14175 ...</span>
<span class="co">#T2D 4 has common genes 16163 ...</span>
<span class="co">#T2D 5 has common genes 15866 ...</span>
<span class="co">#T2D 6 has common genes 15673 ...</span>

<span class="kw">names</span>(Est.prop.Xin)
<span class="co">#[1] "Est.prop.weighted" "Est.prop.allgene"  "Weight.gene"       "r.squared.full"    "Var.prop"  </span></code></pre></div>
<p><a href="../reference/music_prop.html"><code>music_prop</code></a> not only provides MuSiC estimated cell type proportions (<code>Est.prop.weighted</code>), but also non-negative least square(NNLS) estimated proportions (<code>Est.prop.allgene</code>). It also returns the weights for each gene(<code>Weight.gene</code>), R square (<code>r.squared.full</code>) and variance of estimated cell type proportions (<code>Var.prop</code>).</p>
<p>The numeric evaluation is conducted by <a href="../reference/Eval_multi.html"><code>Eval_multi</code></a>, which compares the real and estimated cell type proportions by 1. root-mean-squared deviation (RMSD); 2. mean absolute difference (mAD); 3. Pearson’s correlation (R). The visualization of cell type proportions are provided by <a href="../reference/Prop_comp_multi.html"><code>Prop_comp_multi</code></a>, <a href="../reference/Abs_diff_multi.html"><code>Abs_diff_multi</code></a> and <a href="../reference/Scatter_multi.html"><code>Scatter_multi</code></a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Estimation evaluation</span>

<span class="kw"><a href="../reference/Eval_multi.html">Eval_multi</a></span>(<span class="dt">prop.real =</span> <span class="kw">data.matrix</span>(XinT2D.construct.full$prop.real), 
           <span class="dt">prop.est =</span> <span class="kw">list</span>(<span class="kw">data.matrix</span>(Est.prop.Xin$Est.prop.weighted), 
                           <span class="kw">data.matrix</span>(Est.prop.Xin$Est.prop.allgene)), 
           <span class="dt">method.name =</span> <span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>))

<span class="co">#           RMSD     mAD      R</span>
<span class="co">#MuSiC   0.09881 0.06357 0.9378</span>
<span class="co">#NNLS    0.17161 0.11749 0.8159</span>

<span class="kw">library</span>(cowplot)
prop.comp.fig =<span class="st"> </span><span class="kw"><a href="../reference/Prop_comp_multi.html">Prop_comp_multi</a></span>(<span class="dt">prop.real =</span> <span class="kw">data.matrix</span>(XinT2D.construct.full$prop.real),
                                <span class="dt">prop.est =</span> <span class="kw">list</span>(<span class="kw">data.matrix</span>(Est.prop.Xin$Est.prop.weighted),
                                                <span class="kw">data.matrix</span>(Est.prop.Xin$Est.prop.allgene)),
                                <span class="dt">method.name =</span> <span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>), 
                                <span class="dt">title =</span> <span class="st">'Heatmap of Real and Est. Prop'</span> )

abs.diff.fig =<span class="st"> </span><span class="kw"><a href="../reference/Abs_diff_multi.html">Abs_diff_multi</a></span>(<span class="dt">prop.real =</span> <span class="kw">data.matrix</span>(XinT2D.construct.full$prop.real),
                              <span class="dt">prop.est =</span> <span class="kw">list</span>(<span class="kw">data.matrix</span>(Est.prop.Xin$Est.prop.weighted),
                                              <span class="kw">data.matrix</span>(Est.prop.Xin$Est.prop.allgene)),
                              <span class="dt">method.name =</span> <span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>), 
                              <span class="dt">title =</span> <span class="st">'Abs.Diff between Real and Est. Prop'</span> )

<span class="kw">plot_grid</span>(prop.comp.fig, abs.diff.fig, <span class="dt">labels =</span> <span class="st">"auto"</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">3</span>))</code></pre></div>
<div class="figure">
<img src="images/benchmark.jpg">
</div>
<p>In our <a href="https://www.biorxiv.org/content/early/2018/06/26/354944">manuscript</a>, we also compared our method with existing methods: <a href="https://cibersort.stanford.edu/">CIBERSORT</a> <span class="citation">(see Newman et al. 2015)</span> and <a href="https://github.com/shenorrLab/bseqsc">bseq-sc</a> <span class="citation">(see Baron et al. 2016)</span>.</p>
</div>
<div id="bulk-tissue-cell-type-estimation" class="section level3">
<h3 class="hasAnchor">
<a href="#bulk-tissue-cell-type-estimation" class="anchor"></a>Bulk Tissue Cell Type Estimation</h3>
<p>The deconvolution of 89 subjects from <span class="citation">Fadista et al. (2014)</span> are proformed by bulk data <code>GSE50244.bulk.eset</code> and single cell reference <code>EMTAB.eset</code>. We constrained our estimation on 6 major cell type: alpha, beta, delta, gamma, acinar and ductal, which take up over 90% of the whole islet.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Estimate cell type proportions</span>
Est.prop.GSE50244 =<span class="st"> </span><span class="kw"><a href="../reference/music_prop.html">music_prop</a></span>(<span class="dt">bulk.eset =</span> GSE50244.bulk.eset, <span class="dt">sc.eset =</span> EMTAB.eset, <span class="dt">clusters =</span> <span class="st">'cellType'</span>, <span class="dt">samples =</span> <span class="st">'sampleID'</span>, <span class="dt">select.ct =</span> <span class="kw">c</span>(<span class="st">'alpha'</span>, <span class="st">'beta'</span>, <span class="st">'delta'</span>, <span class="st">'gamma'</span>, <span class="st">'acinar'</span>, <span class="st">'ductal'</span>), <span class="dt">verbose =</span> F)</code></pre></div>
<p>Here we use <a href="../reference/Prop_comp_multi.html"><code>Jitter_Est</code></a> to show the difference between different estimation methods.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Jitter plot of estimated cell type proportions</span>
jitter.fig =<span class="st"> </span><span class="kw"><a href="../reference/Jitter_Est.html">Jitter_Est</a></span>(<span class="kw">list</span>(<span class="kw">data.matrix</span>(Est.prop.GSE50244$Est.prop.weighted),
                             <span class="kw">data.matrix</span>(Est.prop.GSE50244$Est.prop.allgene)),
                        <span class="dt">method.name =</span> <span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>), <span class="dt">title =</span> <span class="st">'Jitter plot of Est Proportions'</span>)

<span class="co"># A more sophisticated jitter plot is provided as below. We seperated the T2D subjects and normal </span>
<span class="co">#subjects by their HbA1c levels.</span>

m.prop.GSE50244 =<span class="st"> </span><span class="kw">rbind</span>(<span class="kw">melt</span>(GSE50244.EMTAB.prop$Est.prop.weighted), 
                        <span class="kw">melt</span>(GSE50244.EMTAB.prop$Est.prop.allgene))

<span class="kw">colnames</span>(m.prop.GSE50244) =<span class="st"> </span><span class="kw">c</span>(<span class="st">'Sub'</span>, <span class="st">'CellType'</span>, <span class="st">'Prop'</span>)
m.prop.GSE50244$CellType =<span class="st"> </span><span class="kw">factor</span>(m.prop.GSE50244$CellType, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">'alpha'</span>, <span class="st">'beta'</span>, <span class="st">'delta'</span>, <span class="st">'gamma'</span>, <span class="st">'acinar'</span>, <span class="st">'ductal'</span>))
m.prop.GSE50244$Method =<span class="st"> </span><span class="kw">factor</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>), <span class="dt">each =</span> <span class="dv">89</span>*<span class="dv">6</span>), <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>))
m.prop.GSE50244$HbA1c =<span class="st"> </span><span class="kw">rep</span>(GSE50244.bulk.eset$hba1c, <span class="dv">2</span>*<span class="dv">6</span>)
m.prop.GSE50244 =<span class="st"> </span>m.prop.GSE50244[!<span class="kw">is.na</span>(m.prop.GSE50244$HbA1c), ]
m.prop.GSE50244$Disease =<span class="st"> </span><span class="kw">factor</span>(<span class="kw">c</span>(<span class="st">'Normal'</span>, <span class="st">'T2D'</span>)[(m.prop.GSE50244$HbA1c &gt;<span class="st"> </span><span class="fl">6.5</span>)+<span class="dv">1</span>], <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">'Normal'</span>, <span class="st">'T2D'</span>))
m.prop.GSE50244$D =<span class="st"> </span>(m.prop.GSE50244$Disease ==<span class="st"> 'T2D'</span>)/<span class="dv">5</span>
m.prop.GSE50244 =<span class="st"> </span><span class="kw">rbind</span>(<span class="kw">subset</span>(m.prop.GSE50244, Disease ==<span class="st"> 'Normal'</span>), <span class="kw">subset</span>(m.prop.GSE50244, Disease !=<span class="st"> 'Normal'</span>))

jitter.new =<span class="st"> </span><span class="kw">ggplot</span>(m.prop.GSE50244, <span class="kw">aes</span>(Method, Prop)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> Method, <span class="dt">color =</span> Disease, <span class="dt">stroke =</span> D, <span class="dt">shape =</span> Disease), 
             <span class="dt">size =</span> <span class="dv">2</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width=</span><span class="fl">0.25</span>, <span class="dt">height=</span><span class="dv">0</span>)) +
<span class="st">  </span><span class="kw">facet_wrap</span>(~<span class="st"> </span>CellType, <span class="dt">scales =</span> <span class="st">'free'</span>) +<span class="st"> </span><span class="kw">scale_colour_manual</span>( <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">'white'</span>, <span class="st">"gray20"</span>)) +
<span class="st">  </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">21</span>, <span class="dv">24</span>))+<span class="st"> </span><span class="kw">theme_minimal</span>()

<span class="kw">plot_grid</span>(jitter.fig, jitter.new, <span class="dt">labels =</span> <span class="st">'auto'</span>)</code></pre></div>
<div class="figure">
<img src="images/GSE50244jitter.jpg">
</div>
<p>Let’s look at the beta cell proportions with HbA1c level.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Create dataframe for beta cell proportions and HbA1c levels</span>
m.prop.ana =<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">pData</span>(GSE50244.bulk.eset)[<span class="kw">rep</span>(<span class="dv">1</span>:<span class="dv">89</span>, <span class="dv">2</span>), <span class="kw">c</span>(<span class="st">'age'</span>, <span class="st">'bmi'</span>, <span class="st">'hba1c'</span>, <span class="st">'gender'</span>)],
                        <span class="dt">ct.prop =</span> <span class="kw">c</span>(Est.prop.GSE50244$Est.prop.weighted[, <span class="dv">2</span>], 
                                    Est.prop.GSE50244$Est.prop.allgene[, <span class="dv">2</span>]), 
                        <span class="dt">Method =</span> <span class="kw">factor</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>), <span class="dt">each =</span> <span class="dv">89</span>), 
                        <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">'MuSiC'</span>, <span class="st">'NNLS'</span>)))
<span class="kw">colnames</span>(m.prop.ana)[<span class="dv">1</span>:<span class="dv">4</span>] =<span class="st"> </span><span class="kw">c</span>(<span class="st">'Age'</span>, <span class="st">'BMI'</span>, <span class="st">'HbA1c'</span>, <span class="st">'Gender'</span>)
m.prop.ana =<span class="st"> </span><span class="kw">subset</span>(m.prop.ana, !<span class="kw">is.na</span>(HbA1c))
m.prop.ana$Disease =<span class="st"> </span><span class="kw">factor</span>( <span class="kw">c</span>(<span class="st">'Normal'</span>, <span class="st">'T2D'</span>)[(m.prop.ana$HbA1c &gt;<span class="st"> </span><span class="fl">6.5</span>) +<span class="st"> </span><span class="dv">1</span>], <span class="kw">c</span>(<span class="st">'Normal'</span>, <span class="st">'T2D'</span>) )
m.prop.ana$D =<span class="st"> </span>(m.prop.ana$Disease ==<span class="st"> 'T2D'</span>)/<span class="dv">5</span>

<span class="kw">ggplot</span>(m.prop.ana, <span class="kw">aes</span>(HbA1c, ct.prop)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">'lm'</span>,  <span class="dt">se =</span> <span class="ot">FALSE</span>, <span class="dt">col =</span> <span class="st">'black'</span>, <span class="dt">lwd =</span> <span class="fl">0.25</span>) +
<span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> Method, <span class="dt">color =</span> Disease, <span class="dt">stroke =</span> D, <span class="dt">shape =</span> Disease), <span class="dt">size =</span> <span class="dv">2</span>, <span class="dt">alpha =</span> <span class="fl">0.7</span>) +<span class="st">  </span><span class="kw">facet_wrap</span>(~<span class="st"> </span>Method) +<span class="st"> </span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">'HbA1c vs Beta Cell Type Proportion'</span>) +<span class="st"> </span><span class="kw">theme_minimal</span>() +<span class="st"> </span>
<span class="st">  </span><span class="kw">scale_colour_manual</span>( <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">'white'</span>, <span class="st">"gray20"</span>)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="dv">21</span>, <span class="dv">24</span>))</code></pre></div>
<div class="figure">
<img src="images/GSE50244beta.jpg">
</div>
</div>
</div>
<div id="estimation-of-cell-type-proportions-with-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#estimation-of-cell-type-proportions-with-clusters" class="anchor"></a>Estimation of cell type proportions with clusters</h2>
<p>Solid tissues often contain closely related cell types, and correlation of gene expression between these cell types leads to collinearity, making it difficult to resolve their relative proportions in bulk data. To deal with collinearity, MuSiC employs a tree-guided procedure that recursively zooms in on closely related cell types. Briefly, we first group similar cell types into the same cluster and estimate cluster proportions, then recursively repeat this procedure within each cluster. At each recursion stage, we only use genes that have low within-cluster variance, a.k.a. the cross-cell consistent genes. This is critical as the mean expression estimates of genes with high variance are affected by the pervasive bias in cell capture of scRNA-seq experiments, and thus cannot serve as reliable reference.</p>
<p>We demonstate this procedure by reproducing the analysis of mouse kidney in MuSiC manuscript.</p>
<div id="data-preparation" class="section level3">
<h3 class="hasAnchor">
<a href="#data-preparation" class="anchor"></a>Data Preparation</h3>
<div id="bulk-data" class="section level4">
<h4 class="hasAnchor">
<a href="#bulk-data" class="anchor"></a>Bulk data</h4>
<p>The dataset <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81492">GEO entry (GSE81492)</a> contains raw RNA-seq and sample annotation data. For the purpose of this vignette, we will use the read counts data <code>Mousebulkeset.rds</code> on the <a href="pages/data.html">data download page</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Download Mouse bulk dataset from Github</span>
Mouse.bulk.eset =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">'https://xuranw.github.io/MuSiC/data/Mousebulkeset.rds'</span>)
Mouse.bulk.eset

<span class="co">#ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">#assayData: 19033 features, 10 samples </span>
<span class="co">#  element names: exprs </span>
<span class="co">#protocolData: none</span>
<span class="co">#phenoData</span>
<span class="co">#  sampleNames: control.NA.27 control.NA.30 ... APOL1.GNA78M (10 total)</span>
<span class="co">#  varLabels: sampleID SubjectName Control</span>
<span class="co">#  varMetadata: labelDescription</span>
<span class="co">#featureData: none</span>
<span class="co">#experimentData: use 'experimentData(object)'</span>
<span class="co">#Annotation:  </span></code></pre></div>
</div>
<div id="single-cell-data" class="section level4">
<h4 class="hasAnchor">
<a href="#single-cell-data" class="anchor"></a>Single cell data</h4>
<p>The single cell data are from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107585">GEO entry (GSE107585)</a>, which contrains read counts for 16273 genes of 43745 cells. A subset of the read counts <code>Mousesubeset.rds</code> are available on the <a href="page/data.html">data download page</a>, in the form of an <code>ExpressionSet</code>. This subset contains 16273 genes and 10000 cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Download EMTAB single cell dataset from Github</span>
Mousesub.eset =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">'https://xuranw.github.io/MuSiC/data/Mousesubeset.rds'</span>)
Mousesub.eset

<span class="co">#ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">#assayData: 16273 features, 10000 samples </span>
<span class="co">#  element names: exprs </span>
<span class="co">#protocolData: none</span>
<span class="co">#phenoData</span>
<span class="co">#  sampleNames: TGGTTCCGTCGGCTCA-2 CGAGCCAAGCGTCAAG-4 ... GAGCAGAGTCAACATC-1 (10000 total)</span>
<span class="co">#  varLabels: sampleID SubjectName cellTypeID cellType</span>
<span class="co">#  varMetadata: labelDescription</span>
<span class="co">#featureData: none</span>
<span class="co">#experimentData: use 'experimentData(object)'</span>
<span class="co">#Annotation:  </span>

<span class="kw">levels</span>(Mousesub.eset$cellType)
<span class="co"># [1] "Endo"     "Podo"     "PT"       "LOH"      "DCT"      "CD-PC"    "CD-IC"    "CD-Trans" "Novel1"  </span>
<span class="co">#[10] "Fib"      "Macro"    "Neutro"   "B lymph"  "T lymph"  "NK"       "Novel2"  </span></code></pre></div>
<p>Notice that the single cell dataset has 16 cell types, including 2 novel cell types and a transition cell type (CD-Trans). We exclude those 3 cell types in our analysis. ## Estimation of cell type proportions</p>
</div>
</div>
<div id="clustering-single-cell-data" class="section level3">
<h3 class="hasAnchor">
<a href="#clustering-single-cell-data" class="anchor"></a>Clustering single cell data</h3>
<p>In <a href="MuSiC.html#estimation-of-cell-type-proportions">previous</a> MuSiC estimation procedure, the first step is producing design matrix, cross-subject mean of relative abundance, cross-subject variance of relative abundance and average library size from single cell reference. <a href="../reference/music_basis.html"><code>music_basis</code></a> is the function that produce the design matrix (<code>Disgn.mtx</code>), cross-subject mean of relative abundance (<code>M.theta</code>), cross-subject variance of relative abundance (<code>Sigma</code>), and average library size (<code>M.S</code>).</p>
<p>MuSiC utilized all genes in the next estimation steps and this will lead to collineary in the design matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Produce the first step information</span>
Mousesub.basis =<span class="st"> </span><span class="kw"><a href="../reference/music_basis.html">music_basis</a></span>(Mousesub.eset, <span class="dt">clusters =</span> <span class="st">'cellType'</span>, <span class="dt">samples =</span> <span class="st">'sampleID'</span>, 
                             <span class="dt">select.ct =</span> <span class="kw">c</span>(<span class="st">'Endo'</span>, <span class="st">'Podo'</span>, <span class="st">'PT'</span>, <span class="st">'LOH'</span>, <span class="st">'DCT'</span>, <span class="st">'CD-PC'</span>, <span class="st">'CD-IC'</span>, <span class="st">'Fib'</span>,
                                           <span class="st">'Macro'</span>, <span class="st">'Neutro'</span>,<span class="st">'B lymph'</span>, <span class="st">'T lymph'</span>, <span class="st">'NK'</span>))

<span class="co"># Plot the dendrogram of design matrix and cross-subject mean of realtive abundance</span>
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
d &lt;-<span class="st"> </span><span class="kw">dist</span>(<span class="kw">t</span>(<span class="kw">log</span>(Mousesub.basis$Disgn.mtx +<span class="st"> </span><span class="fl">1e-6</span>)), <span class="dt">method =</span> <span class="st">"euclidean"</span>)
<span class="co"># Hierarchical clustering using Complete Linkage</span>
hc1 &lt;-<span class="st"> </span><span class="kw">hclust</span>(d, <span class="dt">method =</span> <span class="st">"complete"</span> )
<span class="co"># Plot the obtained dendrogram</span>
<span class="kw">plot</span>(hc1, <span class="dt">cex =</span> <span class="fl">0.6</span>, <span class="dt">hang =</span> -<span class="dv">1</span>, <span class="dt">main =</span> <span class="st">'Cluster log(Design Matrix)'</span>)
d &lt;-<span class="st"> </span><span class="kw">dist</span>(<span class="kw">t</span>(<span class="kw">log</span>(Mousesub.basis$M.theta +<span class="st"> </span><span class="fl">1e-8</span>)), <span class="dt">method =</span> <span class="st">"euclidean"</span>)
<span class="co"># Hierarchical clustering using Complete Linkage</span>
<span class="co"># hc2 &lt;- hclust(d, method = "complete" )</span>
hc2 &lt;-<span class="st"> </span><span class="kw">hclust</span>(d, <span class="dt">method =</span> <span class="st">"complete"</span>)
<span class="co"># Plot the obtained dendrogram</span>
<span class="kw">plot</span>(hc2, <span class="dt">cex =</span> <span class="fl">0.6</span>, <span class="dt">hang =</span> -<span class="dv">1</span>, <span class="dt">main =</span> <span class="st">'Cluster log(Mean of RA)'</span>)</code></pre></div>
<div class="figure">
<img src="images/Cluster.jpg">
</div>
<p>The immune cells are clustered together and so are the epithelial cells. Notice that DCT and PT are close to each other. The cut-off is user determined. Here we cut 13 cell types into 4 groups:</p>
<ul>
<li>Group 1: Neutro</li>
<li>Group 2: Podo</li>
<li>Group 3: Endo, CD-PC, CD-IC, LOH, DCT, PT</li>
<li>Group 4: Fib, Macro, NK, B lymph, T lymph</li>
</ul>
<p>The tree-guided recursive estimation for mouse kidney analysis includes 2 steps:</p>
<ul>
<li>Step 1. Estimate proportions of each cluster; <span class="math inline">\((p_1,p_2,p_3,p_4)\)</span>
</li>
<li>Step 2. Estimate cell type proportions within each clusters; <span class="math inline">\((p_{31},p_{32},.,p_{36},p_{41},.,p_{45})\)</span>
</li>
</ul>
<p>We manually specify the cluster and annotated single cell data with cluster information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clusters.type =<span class="st"> </span><span class="kw">list</span>(<span class="dt">C1 =</span> <span class="st">'Neutro'</span>, <span class="dt">C2 =</span> <span class="st">'Podo'</span>, <span class="dt">C3 =</span> <span class="kw">c</span>(<span class="st">'Endo'</span>, <span class="st">'CD-PC'</span>, <span class="st">'LOH'</span>, <span class="st">'CD-IC'</span>, <span class="st">'DCT'</span>, <span class="st">'PT'</span>), <span class="dt">C4 =</span> <span class="kw">c</span>(<span class="st">'Macro'</span>, <span class="st">'Fib'</span>, <span class="st">'B lymph'</span>, <span class="st">'NK'</span>, <span class="st">'T lymph'</span>))

cl.type =<span class="st"> </span><span class="kw">as.character</span>(Mousesub.eset$cellType)

for(cl in <span class="dv">1</span>:<span class="kw">length</span>(clusters.type)){
  cl.type[cl.type %in%<span class="st"> </span>clusters.type[[cl]]] =<span class="st"> </span><span class="kw">names</span>(clusters.type)[cl]
}
<span class="kw">pData</span>(Mousesub.eset)$clusterType =<span class="st"> </span><span class="kw">factor</span>(cl.type, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="kw">names</span>(clusters.type), <span class="st">'CD-Trans'</span>, <span class="st">'Novel1'</span>, <span class="st">'Novel2'</span>))

<span class="co"># 13 selected cell types</span>
s.mouse =<span class="st"> </span><span class="kw">unlist</span>(clusters.type)
s.mouse
<span class="co">#       C1        C2       C31       C32       C33       C34       C35       C36       C41       C42 </span>
<span class="co"># "Neutro"    "Podo"    "Endo"   "CD-PC"     "LOH"   "CD-IC"     "DCT"      "PT"   "Macro"     "Fib" </span>
<span class="co">#      C43       C44       C45 </span>
<span class="co">#"B lymph"      "NK" "T lymph" </span></code></pre></div>
<p>We then select genes that differential expressed genes within cluster <code>C3</code> (Epithelial cells) and <code>C4</code> (Immune cells), available on <a href="pages/data.html">data download page</a>. Function <code>music_prop.cluster</code> is used for estimation with cluster.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">'https://xuranw.github.io/MuSiC/data/IEmarkers.RData'</span>)

Est.mouse.bulk =<span class="st"> </span><span class="kw"><a href="../reference/music_prop.cluster.html">music_prop.cluster</a></span>(<span class="dt">bulk.eset =</span> Mouse.bulk.eset, <span class="dt">sc.eset =</span> Mousesub.eset, <span class="dt">group.markers =</span> IEmarkers, <span class="dt">clusters =</span> <span class="st">'cellType'</span>, <span class="dt">group =</span> <span class="st">'clusterType'</span>, <span class="dt">samples =</span> <span class="st">'sampleID'</span>, <span class="dt">clusters.type =</span> clusters.type)</code></pre></div>
<p>The results might be different from the one presented in the manuscript due to incomplete reference single cell dataset.</p>
</div>
</div>
</div>
<div id="reference" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h1>
<div id="refs" class="references">
<div id="ref-baron2016single">
<p>Baron, Maayan, Adrian Veres, Samuel L Wolock, Aubrey L Faust, Renaud Gaujoux, Amedeo Vetere, Jennifer Hyoje Ryu, et al. 2016. “A Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter-and Intra-Cell Population Structure.” <em>Cell Systems</em> 3 (4). Elsevier: 346–60.</p>
</div>
<div id="ref-beckerman2017transgenic">
<p>Beckerman, Pazit, Jing Bi-Karchin, Ae Seo Deok Park, Chengxiang Qiu, Patrick D Dummer, Irfana Soomro, Carine M Boustany-Kari, et al. 2017. “Transgenic Expression of Human Apol1 Risk Variants in Podocytes Induces Kidney Disease in Mice.” <em>Nature Medicine</em> 23 (4). Nature Publishing Group: 429.</p>
</div>
<div id="ref-fadista2014global">
<p>Fadista, João, Petter Vikman, Emilia Ottosson Laakso, Inês Guerra Mollet, Jonathan Lou Esguerra, Jalal Taneera, Petter Storm, et al. 2014. “Global Genomic and Transcriptomic Analysis of Human Pancreatic Islets Reveals Novel Genes Influencing Glucose Metabolism.” <em>Proceedings of the National Academy of Sciences</em> 111 (38). National Acad Sciences: 13924–9.</p>
</div>
<div id="ref-newman2015robust">
<p>Newman, Aaron M, Chih Long Liu, Michael R Green, Andrew J Gentles, Weiguo Feng, Yue Xu, Chuong D Hoang, Maximilian Diehn, and Ash A Alizadeh. 2015. “Robust Enumeration of Cell Subsets from Tissue Expression Profiles.” <em>Nature Methods</em> 12 (5). Nature Publishing Group: 453.</p>
</div>
<div id="ref-park2018single">
<p>Park, Jihwan, Rojesh Shrestha, Chengxiang Qiu, Ayano Kondo, Shizheng Huang, Max Werth, Mingyao Li, Jonathan Barasch, and Katalin Suszták. 2018. “Single-Cell Transcriptomics of the Mouse Kidney Reveals Potential Cellular Targets of Kidney Disease.” <em>Science</em>. American Association for the Advancement of Science, eaar2131.</p>
</div>
<div id="ref-segerstolpe2016single">
<p>Segerstolpe, Åsa, Athanasia Palasantza, Pernilla Eliasson, Eva-Marie Andersson, Anne-Christine Andréasson, Xiaoyan Sun, Simone Picelli, et al. 2016. “Single-Cell Transcriptome Profiling of Human Pancreatic Islets in Health and Type 2 Diabetes.” <em>Cell Metabolism</em> 24 (4). Elsevier: 593–607.</p>
</div>
<div id="ref-xin2016rna">
<p>Xin, Yurong, Jinrang Kim, Haruka Okamoto, Min Ni, Yi Wei, Christina Adler, Andrew J Murphy, George D Yancopoulos, Calvin Lin, and Jesper Gromada. 2016. “RNA Sequencing of Single Human Islet Cells Reveals Type 2 Diabetes Genes.” <em>Cell Metabolism</em> 24 (4). Elsevier: 608–15.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#data">Data</a></li>
      </ul>
</li>
      <li><a href="#music-deconvolution">MuSiC Deconvolution</a></li>
      <li>
<a href="#sample-analysis">Sample Analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#data-summary">Data Summary</a></li>
      <li><a href="#estimation-of-cell-type-proportions">Estimation of cell type proportions</a></li>
      <li><a href="#estimation-of-cell-type-proportions-with-clusters">Estimation of cell type proportions with clusters</a></li>
      </ul>
</li>
      <li><a href="#reference">Reference</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Xuran Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
